protocols/kad/src/query/peers/closest.rs

```diff
@@ -55,6 +57,7 @@ pub struct ClosestPeersIterConfig {
     /// The `α` parameter in the Kademlia paper. The maximum number of peers that
     /// the iterator is allowed to wait for in parallel while iterating towards the closest
     /// nodes to a target. Defaults to `ALPHA_VALUE`.
+    // TODO: Should this be configurable?
```

Comment:
From past open source maintenance experience I prefer small configuration surfaces because:

- They are usually more difficult to misuse.

- They are easier to maintain.

- They are easier to change in the future.

As far as I know noone asked this to be configurable in the past. Thus I would not make it configurable now, but only once someone needs it to be configurable.

What do you think?

Reply:
I'm a bit confused, because:

  a) This is not (crate-)public configuration (or is it?).
  b) At least for the transition period, these iterators need to be configured with a different parallelism in the context of disjoint vs standard Kademlia queries, no? And furthermore, if we eventually end up only supporting disjoint-path queries, the whole concept of `parallelism` inside a single `ClosestPeersIter` could be removed, not just the configuration option, unless we with to make that configurable.

Reply:
Ah, I think my `TODO` comment is confusing, sorry about that. Let me rephrase:

`// TODO: Should parallelism be configurable from outside the crate, e.g. by a user?`

Do you agree that `parallelism` should not be configurable from outside of the crate @romanb for now?

> At least for the transition period, these iterators need to be configured with a different parallelism in the context of disjoint vs standard Kademlia queries, no?

Yes. From within the crate it is configurable and should stay configurable  given that `ClosestDisjointPeersIter` makes use of it.

> And furthermore, if we eventually end up only supporting disjoint-path queries, the whole concept of parallelism inside a single ClosestPeersIter could be removed, not just the configuration option, unless we wish to make that configurable.

:+1: Agreed.

Reply:
> Do you agree that parallelism should not be configurable from outside of the crate @romanb for now?

Well, I think with the introduction of disjoint paths it should be, because the parallelism controls the number of disjoint paths. What is confusing me is why you put this TODO on the internal configuration of a `ClosestPeersIter`, because clearly a) that is not in the public API and b) as we agreed it needs to be configurable for now at this level until we eventually only support disjoint path queries.